tests/robustness/coverage/key_pattern_test.go

```diff
@@ -0,0 +1,173 @@
+// Copyright 2025 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package coverage_test
+
+import (
+	"strings"
+	"testing"
+)
+
+func pattern(k string) (string, bool) {
+	if k == "/registry/health" || k == "compact_rev_key" {
+		return k, true
+	}
+	k, valid := strings.CutPrefix(k, "/registry")
+	if !valid {
+		return k, false
+	}
+	slash := ""
+	if strings.HasSuffix(k, "/") {
+		slash = "/"
+	}
+	s := strings.Split(strings.Trim(k, "/"), "/")
+	if len(s[0]) == 0 {
+		return k, false
+	}
+
+	resource, shift := resourceOrCRD(s[0])
```

Comment:
Isn't `resource` really a `apiResource` where api can be empty for non core?

Reply:
Renamed to `apiResource`